src/channels.js

```diff
@@ -1,5 +1,67 @@
 const log = require('./logger');
 
+/**
+ * Join a channel given a user and connection.
+ */
+const joinChannels = (app, user, connection) => {
+    // Keep a reference to all authenticated users.
+    app.channel('authenticated').join(connection);
+
+    // Subscribe user to all attended workshops.
+    // Assumes that the workshop assignment is stored
+    // on an array of the user.
+    user.workshops.forEach(workshopId =>
+        app.channel(`workshops/${workshopId}`).join(connection)
+    );
+
+    if (user.permissions.includes('admin')) {
+        app.channel('admins').join(connection);
+    }
+
+    if (user.permissions.includes('student')) {
+        app.channel('students').join(connection);
+    }
+}
+
+/**
+ * Get a user to leave all channels.
+ */
+const leaveChannels = (app, user) => {
+    app.channel(app.channels).leave((connection) => {
+        connection = fixConnection(connection);
+        return connection.user.username === user.username
+    });
+};
+
+/**
+ * Leave and re-join all channels with new user information.
+ */
+const updateChannels = (app, user) => {
+    // Find all connections for this user.
+    const { connections } = app.channel(app.channels).filter(connection =>
+        connection.user.username === user.username
+    );
+
+    // Leave all channels
+    leaveChannels(app, user);
+
+    // Re-join all channels with the updated user information
+    connections.forEach(connection => joinChannels(user, connection));
+}
+
+/**
+ * XXX: Fix for https://github.com/feathersjs/feathers/issues/941
+ * This was addressed in version 4.0.0, which is currently in 
+ * pre-release. Update when release is stable.
+ */
+function fixConnection(connection) {
+    if (!connection.user && Object.getOwnPropertySymbols(connection).length > 0) {
```

Comment:
You could assign `Object.getOwnPropertySymbols(connection)` to some `const` - it's quite long and you use it twice.